File: src/1_100/0006_ZigZag_Conversion/ZigZag_Conversion.java
class Solution {
    public static String convert(String s, int numRows) {
        if(s==null || numRows==1) {
            return s;
        }

        int size = Math.min(s.length(),numRows);
        StringBuilder[] arr = new StringBuilder[size];
        for(int i=0;i<size;i++) {
            arr[i] = new StringBuilder();
        }
        int i = 0;
        int n = s.length();
        while(i < n) {
            for (int j = 0; j < numRows; j++) {
                if (i < n) {
                    char c = s.charAt(i);
                    arr[j].append(c);
                    i++;
                }
            }
            for (int j = numRows - 2; j > 0; j--) {
                if (i < n) {
                    char c = s.charAt(i);
                    arr[j].append(c);
                    i++;
                }
            }
        }
        for(int k=1;k<arr.length;k++) {
            arr[0].append(arr[k]);
        }
        return arr[0].toString();
    }
}

File: src/1_100/0006_ZigZag_Conversion/ZigZag_Conversion.py
class Solution(object):
	def convert(self, s, numRows):
		if not s or numRows<=1:
			return s
		n = len(s)		
		res = [[] for _ in xrange(min(n,numRows))]	
		is_going_down = False
		j = 0
		ans = ""
		for c in s:
			res[j].append(c)
			if j==0 or j==numRows-1:
				is_going_down = not is_going_down
			j = j+1 if is_going_down else j-1
		for i in res:
			ans += "".join(i)
		return ans